src/app/modules/shopping-cart/components/containers/shopping-cart-details/shopping-cart-details.component.ts

```diff
@@ -1,12 +1,58 @@
+import { HttpClient } from '@angular/common/http';
 import { Component } from '@angular/core';
-import { productsMockList } from 'src/app/components/mocks/products.mock';
-import { Product } from 'src/app/modules/shared/types/product';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { Observable, of } from 'rxjs';
+import { ShoppingCartService } from 'src/app/modules/shared/services/shopping.cart.service';
+import { OrderProductWithId } from 'src/app/modules/shared/types/order.product';
+import { OrderProduct } from 'src/app/modules/shared/types/product.order';
+import { OrderByCustomer } from 'src/app/modules/shared/types/product.order.with.customer';
+import { environment } from 'src/environments/environment';
 
+@UntilDestroy({})
 @Component({
   selector: 'app-shopping-cart-details',
   templateUrl: './shopping-cart-details.component.html',
   styleUrls: ['./shopping-cart-details.component.scss'],
 })
 export class ShoppingCartDetailsComponent {
-  productList: Product[] = productsMockList;
+  private shoppingCartService: ShoppingCartService;
+  productList: OrderProduct[];
+  constructor(
+    shoppingCartService: ShoppingCartService,
+    private http: HttpClient
+  ) {
+    this.shoppingCartService = shoppingCartService;
+    this.productList = this.shoppingCartService.getShoppingCart();
+  }
+
+  getProducts(): Observable<OrderProduct[]> {
+    return of((this.productList = this.shoppingCartService.getShoppingCart()));
+  }
+
+  catchProductDelete(productId: string) {
+    this.productList = this.shoppingCartService.removeFromShoppingCart(
+      this.productList,
+      productId
+    );
+  }
+
+  catchCheckoutOrder(productList: OrderProduct[]) {
+    const dummyCustomer = 'bacb89d7-9070-426f-a662-facb538c8da9';
+    const orderList: OrderProductWithId[] = [];
+    productList.forEach((product) => {
+      orderList.push({
+        productId: product.id,
+        quantity: product.quantity,
+      });
+    });
+    const order: OrderByCustomer = {
+      products: orderList,
+      customerId: dummyCustomer,
+    };
+    this.http
+      .post(`${environment.apiUrl}/orders`, order)
+      .pipe(untilDestroyed(this))
+      .subscribe();
+    alert('order created successfully');
```

Comment:
Nothing seems to be done in this subscription. Maybe would make more sense if you moved the alert here in case of success. It could happen that your request is not successfully, but the user would still get the success message because it does not wait for the response